ratelimit.go

```diff
@@ -33,12 +34,37 @@ type Bucket struct {
 	availTick int64
 }
 
+// Clock is used to inject testable fakes.
+type Clock interface {
```

Comment:
Hmm... this is the first time that I have looked at this code in detail.

The Clock interface that Juju uses elsewhere is a subset of this one:
https://github.com/juju/utils/blob/master/clock/clock.go

So, instead of Sleep, we generally use After. Mostly because we have selects with cancellation channels.